api/src/app/modules/external/fsp-tracker/fsp-tracker.controller.ts

```diff
@@ -0,0 +1,42 @@
+import { ProjectFindCriteria } from '@api-modules/project/project.service';
+import { ArgumentMetadata, BadRequestException, Controller, Get, HttpStatus, Injectable, PipeTransform, Query } from '@nestjs/common';
+import { ApiQuery, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { FomFspTrackerResponse } from '@src/app/modules/external/fsp-tracker/fsp-tracker.dto';
+import { FspTrackerService } from '@src/app/modules/external/fsp-tracker/fsp-tracker.service';
+import { PinoLogger } from 'nestjs-pino';
+
+// Custom pipe transformer for controller parameter conversion.
+@Injectable()
+export class PositiveIntPipe implements PipeTransform<number, number> {
+    transform(value: any, metadata: ArgumentMetadata) {
+
+        if(/^\d+$/.test(value)) {
+            const intValue = parseInt(value);
+            if ( intValue > 0) {
+                return value;
+            }
+        }
+
+        throw new BadRequestException('Value must be positive integer.');
+    }
+}
+
+@ApiTags("external")
+@Controller("external")
+export class FspTrackerController {
+  constructor(
+    private readonly service: FspTrackerService,
+    private readonly _logger: PinoLogger) {
+  }
+
+  @Get("fsp-tracker")
+  @ApiQuery({ name: 'fspId', required: true})
+  @ApiResponse({ status: HttpStatus.OK, type: [FomFspTrackerResponse] })
```

Comment:
Hi @basilv as discussed, this API is separated into different controller.
I intentionally make it into a folder structure like this: "api/src/app/modules/**external**/**fsp-tracker**"; and placed the "**fsp-tracker module**" as a submodule under "**external**" module.
And the endpoint path also look like this (GET api/**external**/**fsp-tracker**):
![image](https://github.com/user-attachments/assets/254e7d74-0b99-4c6a-8475-ecaad3ff6aa1)

Welcome to suggest different name or if you don't feel "**external**" folder/module is needed then we can change.
(For "external", the intention was for all other external services, like FSP Tracker, to interface with FOM be under a common module)
Thank you!

Reply:
Hi Basil, for folder structure, I will use "modules/external/projects-by-fsp", but for endpoint path, the FSP Tracker team probably would be confused by what "**project**" means... would it be using "**fom**" at endpoint path "GET **api/external/fom-by-fsp**" or ""GET **api/external/fom** (query parameter is on **fspId** already)" make more sense to them?

Reply:
Hmm, yes /fom makes more sense than project.